pkg/controller/upgrade.go

```diff
@@ -21,53 +21,100 @@ package controller
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"time"
 
+	"github.com/sirupsen/logrus"
+
 	"github.com/google/go-github/v44/github"
 
-	"github.com/infracloudio/botkube/pkg/log"
 	"github.com/infracloudio/botkube/pkg/notify"
 	"github.com/infracloudio/botkube/pkg/version"
 )
 
-var (
-	notified          = false
-	botkubeUpgradeMsg = "Newer version (%s) of BotKube is available :tada:. Please upgrade BotKube backend.\n" +
-		"Visit botkube.io for more info."
+const (
+	defaultDuration = 24 * time.Hour
+	upgradeMsgFmt   = "Newer version (%s) of BotKube is available :tada:. Please upgrade BotKube backend.\nVisit botkube.io for more info."
+	repoOwner       = "infracloudio"
+	repoName        = "botkube"
 )
 
-func checkRelease(notifiers []notify.Notifier) {
-	ctx := context.Background()
-	client := github.NewClient(nil)
-	release, _, err := client.Repositories.GetLatestRelease(ctx, "infracloudio", "botkube")
-	if err == nil {
-		log.Debugf(fmt.Sprintf("Upgrade notifier:: latest release info=%+v", release))
-		if release.TagName == nil {
-			return
-		}
+// GitHubRepoClient describes the client for getting latest release for a given repository.
+type GitHubRepoClient interface {
+	GetLatestRelease(ctx context.Context, owner, repo string) (*github.RepositoryRelease, *github.Response, error)
+}
 
-		// Send notification if newer version available
-		if version.Short() != *release.TagName {
-			sendMessage(notifiers, fmt.Sprintf(botkubeUpgradeMsg, *release.TagName))
-			notified = true
-		}
-	}
+// UpgradeChecker checks for new BotKube releases.
+type UpgradeChecker struct {
```

Comment:
I agree, but I wanted to keep the current behavior. For now, I will log the error from `notified, err := c.notifyAboutUpgradeIfShould(ctx)` instead of returning 👍

Reply:
yep, as mentioned I just wanted to start the discussion and if you agree, I will create a dedicated issue to track that 👍

Reply:
Awesome 👍 please do - or tell me if I should create one 🙂